### DISCUSSION TITLE
Convert pdf greyscale to B&W

### DISCUSSION THREAD

**Discussion by flywire** [ImageMagick/ImageMagick#6100] [Help]
### ImageMagick version

7.1.0-Q16

### Operating system, version and so on

Win 10

### Description

Trying to convert [test.pdf](https://github.com/ImageMagick/ImageMagick/files/10809312/test.pdf) from greyscale to B&W using `magick "e:test.pdf" -threshold 75% e:new.pdf`. Output is outlining objects rather than converting them to black.

![image](https://user-images.githubusercontent.com/11288701/220787711-20aa00b2-f64e-46f7-9d86-257c20aeca1d.png)

![image](https://user-images.githubusercontent.com/11288701/220787755-5826c15e-30f8-49b6-847f-62b26cae9843.png)

[new.pdf](https://github.com/ImageMagick/ImageMagick/files/10809136/new.pdf)

---

Edit: Added [test.pdf](https://github.com/ImageMagick/ImageMagick/files/10809312/test.pdf)

**Comment by fmw42** [comment]
This is what I get using IM 7.1.0.62 on Mac OSX Monterey:

`magick 220787711-20aa00b2-f64e-46f7-9d86-257c20aeca1d.png -colorspace gray -auto-threshold otsu 220787711-20aa00b2-f64e-46f7-9d86-257c20aeca1d.pdf`

[220787711-20aa00b2-f64e-46f7-9d86-257c20aeca1d.pdf](https://github.com/ImageMagick/ImageMagick/files/10809211/220787711-20aa00b2-f64e-46f7-9d86-257c20aeca1d.pdf)


Here is a png version of the output:

![220787711-20aa00b2-f64e-46f7-9d86-257c20aeca1d_thresh](https://user-images.githubusercontent.com/13458218/220790383-081beb0b-cde8-44e7-abd1-64303ac27f6e.png)

**Reply by flywire** [reply]
My bad, I hadn't added [test.pdf](https://github.com/ImageMagick/ImageMagick/files/10809312/test.pdf).

I'm still getting the same result with `magick "e:test.pdf" -colorspace gray -auto-threshold otsu e:new.pdf`. Do you think it is a bug with Win10? I need to retore the dpi to 600 or 1200 too.

**Comment by fmw42** [comment]
Your PDF has an alpha channel. Remove it. So try

`magick test.pdf -alpha off -colorspace gray -auto-threshold otsu test_new.pdf`

[test_new.pdf](https://github.com/ImageMagick/ImageMagick/files/10809367/test_new.pdf)

**Comment by flywire** [comment]
> ImageMagick[®](http://tarr.uspto.gov/servlet/tarr?regser=serial&entry=78333969) is a free and [open-source](https://github.com/ImageMagick/ImageMagick/discussions/www/license.html) software suite for displaying, converting, and editing raster image and vector image files.

Is ImageMagick converting the vector pdf to a 75dpi raster image and embedding it in the pdf? In this case that is fine.

**Reply by fmw42** [reply]
Yes. Imagemagick is a raster processor. If you need vectors, I believe you can load potrace on your system and IM will use it to output vectors, though they may not be in the form your want. 

The default density is 72 dpi.  If you want to increase density, then add -density XX where XX=300 or whatever you want before your output pdf.

**Comment by flywire** [comment]
`magick -density 300 test.pdf -alpha off -colorspace gray -auto-threshold otsu new.pdf`


[new.pdf](https://github.com/ImageMagick/ImageMagick/files/10809577/new.pdf)

![image](https://user-images.githubusercontent.com/11288701/220802149-dc119821-c8b5-40e5-863d-f6331c4d7cce.png)

**Comment by fmw42** [comment]
I must have expressed it wrongly. That increased the density by 4 by reducing the page size by 4 in each direction. I want to increase the sampling of the original vector image by 4 and retain the A4 page size.

**Reply by fmw42** [reply]
To increase density and save as A4, try

`magick -density 288 test.pdf -alpha off -colorspace gray -auto-threshold otsu -resize "%[papersize:A4]" -gravity center -extent "%[papersize:A4]" -density 72 test_new.pdf
`

[test_new.pdf](https://github.com/ImageMagick/ImageMagick/files/10809601/test_new.pdf)

**Reply by flywire** [reply]
The command above with `-density 288` didn't give me the resolution I was after:

![image](https://github.com/ImageMagick/ImageMagick/assets/11288701/4ed71d6c-a0a0-48bd-a7c6-568b2e4c88bf)

but it was enough for me to work out how to get the following last year:

![image](https://github.com/ImageMagick/ImageMagick/assets/11288701/d8d38aa1-2c8a-413f-9fa4-e6fd2ffb1a39)

I don't understand how I generated the higher resolution last year. Increasing the density seems to generate the same low resolution file. Any thoughts?

**Comment by flywire** [comment]
Better to hack the file directly for best output:

1. Standard `CMYK` colors to Black
  * `/C1[0.0 0.95 1.0 0.0]` to `/C1[0.0 0.00 0.0 1.0]` ie Purple to Black
  * `/C1[0.0 0.24 0.94 0.0]` to  `/C1[0.0 0.00 0.0 0.3]` ie Yellow to Gray
2. Remove standard shading
  * `/SA true/SMask/None/Type/ExtGState/ca 1.0/op false` to `/SA true/SMask/None/Type/ExtGState/ca 0.0/op false`

Note: This removes number on barcode and there will be no shading outline for first TFN.

**Comment by flywire** [comment]
See #8151
